libraries/Microsoft.Bot.PublishValidation/Helpers/BotValidatorHelper.cs

```diff
@@ -0,0 +1,377 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.PublishValidation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using Microsoft.Bot.Configuration;
+
+    public class BotValidatorHelper
+    {
+        private const string FORBIDDEN_ENDPOINTS = "Forbidden Endpoints";
+        private const string REQUIRED_ENDPOINTS = "Required Endpoints";
+
+        /// <summary>
+        /// Checks if a bot file is valid or not according to some configuration options
+        /// </summary>
+        /// <param name="configurationOptions"></param>
+        /// <param name="messages"></param>
+        /// <returns></returns>
+        public static bool BotFileIsValid(ConfigurationOptions configurationOptions, ref IEnumerable<NotificationMessage> messages)
+        {
+            try
+            {
+                // Load the first .bot file from the provided folder.
+                // Also validates its existence (throws an exception is there isn't any file)
+                var botConfiguration = BotValidatorHelper.LoadFromFolder(configurationOptions.ProjectPath);
+
+                return BotValidatorHelper.ValidateBotFile(botConfiguration, configurationOptions, ref messages);
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+        }
+
+        /// <summary>
+        /// Performs the validation of the specified .bot file, according the specified validation options
+        /// </summary>
+        /// <param name="botConfiguration"></param>
+        /// <param name="options"></param>
+        /// <param name="messages"></param>
+        /// <returns></returns>
+        private static bool ValidateBotFile(BotConfiguration botConfiguration, ConfigurationOptions options, ref IEnumerable<NotificationMessage> messages)
+        {
```

Comment:
Not clear why this is a ref parameter, or an IEnumerable. 

Seems like this should just be a List<NotificationMessage>, and passes as a standard parameter.

Reply:
We already updated this method to use a List.